app/src/main/java/com/jerboa/ui/components/drawer/Drawer.kt

```diff
@@ -0,0 +1,414 @@
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Add
+import androidx.compose.material.icons.outlined.Bookmarks
+import androidx.compose.material.icons.outlined.Close
+import androidx.compose.material.icons.outlined.ExpandLess
+import androidx.compose.material.icons.outlined.ExpandMore
+import androidx.compose.material.icons.outlined.LocationCity
+import androidx.compose.material.icons.outlined.Login
+import androidx.compose.material.icons.outlined.Person
+import androidx.compose.material.icons.outlined.Public
+import androidx.compose.material.icons.outlined.Settings
+import androidx.compose.material3.Divider
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.navigation.NavController
+import androidx.navigation.compose.rememberNavController
+import com.jerboa.R
+import com.jerboa.api.ApiState
+import com.jerboa.datatypes.samplePerson
+import com.jerboa.datatypes.types.Community
+import com.jerboa.datatypes.types.GetSiteResponse
+import com.jerboa.datatypes.types.ListingType
+import com.jerboa.datatypes.types.MyUserInfo
+import com.jerboa.datatypes.types.Person
+import com.jerboa.db.Account
+import com.jerboa.db.AccountViewModel
+import com.jerboa.federatedNameShown
+import com.jerboa.ui.components.common.IconAndTextDrawerItem
+import com.jerboa.ui.components.common.LargerCircularIcon
+import com.jerboa.ui.components.common.PictrsBannerImage
+import com.jerboa.ui.components.common.simpleVerticalScrollbar
+import com.jerboa.ui.components.common.toLogin
+import com.jerboa.ui.components.community.CommunityLinkLarger
+import com.jerboa.ui.components.home.NavTab
+import com.jerboa.ui.components.person.PersonName
+import com.jerboa.ui.theme.DRAWER_BANNER_SIZE
+import com.jerboa.ui.theme.LARGE_PADDING
+import com.jerboa.ui.theme.SMALL_PADDING
+import com.jerboa.ui.theme.XL_PADDING
+import com.jerboa.ui.theme.muted
+
+@Composable
+fun Drawer(
+    siteRes: ApiState<GetSiteResponse>,
```

Comment:
Just have this pass in `myUserinfo?` . I've tried hard to avoid passing the ApiStates to anything but the top-level screens, and especially not to the lower-level composables.

Reply:
That was there before, i just moved it to a new location, but i have changed it.